wallet/walletrpc.c

```diff
@@ -341,22 +359,36 @@ static void json_newaddr(struct command *cmd,
 
 	txfilter_add_derkey(cmd->ld->owned_txfilter, ext.pub_key);
 
-	redeemscript = bitcoin_redeem_p2sh_p2wpkh(cmd, &pubkey);
-	sha256(&h, redeemscript, tal_count(redeemscript));
-	ripemd160(&p2sh, h.u.u8, sizeof(h));
+	if (is_p2wpkh) {
+		/* segwit_addr_encode's internal buffer size is 65, so just use that */
+		out = tal_arr(cmd, char, 65);
```

Comment:
Better to follow the header comments of segwit_addr_encode: https://github.com/ElementsProject/lightning/blob/master/common/bech32.h#L34

    out = tal_arr(cmd, char, 73 + strlen(get_chainparams(cmd->id)->bip173_name);

Reply:
👍